front/src/features/filter/hooks/useTimeDropdown.ts

```diff
@@ -1,15 +1,12 @@
 import { useNavigation } from "@/hooks/useNavigation";
-import { safeCastToTimeOption } from "@/types/safeCast";
 import { useState } from "react";
 import { getSelectedTimeFromURL } from "../helpers/getSelectedTimeFromURL";
+import { timeOptionConstants } from "@/services/helpers/timeConstants";
+import { displayNameToTimeOption } from "@/services/helpers/displayNameToTimeOption";
 
 export const useTimeDropdown = () => {
-    const timeOptions = [
-        "Last hour",
-        "Last day",
-        "Last week",
-        "Custom interval",
-    ];
+    let timeOptions = timeOptionConstants.map((option) => option.displayName);
```

Comment:
is it possible to do
```suggestion
    const timeOptions = timeOptionConstants.map((option) => option.displayName).concat("Custom interval");
```

Is there a reason for using let rather than const?

Reply:
The reason was to separate the two steps of the timeOptions' construction. See new commit for refactor